api/v1beta2/gitrepository_types.go

```diff
@@ -170,11 +166,17 @@ type GitRepositoryRef struct {
 	// +optional
 	SemVer string `json:"semver,omitempty"`
 
+	// Name of the reference to check out; takes precedence over Branch, Tag and SemVer.
+	//
+	// It must be a valid Git reference: https://git-scm.com/docs/git-check-ref-format#_description
+	// Examples: "refs/heads/main", "refs/tags/v0.1.0", "refs/pull/420/head", "refs/merge-requests/1/head"
```

Comment:
Let's add here a kubebuilder regex that matches `refs/<any>/<any>`

Reply:
The validation is much more complex than just `refs/...`: https://git-scm.com/docs/git-check-ref-format#_description

Reply:
I would rather just let the Git implementation figure out if the arbitrary reference is correct.

Reply:
we include the ref name in the error message: https://github.com/fluxcd/pkg/blob/main/git/gogit/clone.go#L372 and we have some basic validation: https://github.com/fluxcd/pkg/blob/main/git/gogit/clone.go#L402-L404. given these, i don't think we have a pressing need to validate the ref name at the API level